third_party/blink/public/mojom/cobalt/crash_annotator/crash_annotator.mojom

```
@@ -0,0 +1,22 @@
+// Copyright 2024 The Cobalt Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+module blink.mojom;
+
+// The browser process must provide an implementation of this interface so that
+// the renderer process can implement the CrashAnnotator Blink API.
+interface CrashAnnotator {
+  // Recieves an annotation to set and responds with the result.
+  SetString(string key, string value) => (bool result);
+};
```

Any reason not to name this the same as the current h5vcc crash log api?

---

See https://github.com/youtube/cobalt/pull/4574#discussion_r1892971477, when Holden gets back I think he's going to work on changing this but just wanted to get something in before vacation

---

Sweet! Thanks. Ya, I suspect trying to rename this will make things much more difficult on the Kabuki side. While not impossible, backwards compatibility is easier if the api name remains stable.

---

> Any reason not to name this the same as the current h5vcc crash log api?

The simple reason is that I'm not a huge fan of the "CrashLog" name and this migration seemed like a decent time to do some renaming. To me it makes more sense to think about a "CrashAnnotator" object instead of a "CrashLog" object

But yeah if the renaming will be a pain for Kabuki then I'm fine keeping the status quo. I've asked that question in go/h5vcc-chrobalt, so let's see what the Kabuki team says.

Also fwiw it should be fine to name things slightly differently in the Blink vs. Mojo layers, although it could cause a bit of confusion.

---

I agree, the whole h5vcc naming convention is not ideal. If we do rename though, what'll happen in Kabuki code is someone will need to do an `if` check somewhere based on Cobalt version. If this version use this api, otherwise, this api. Not the end of the world, but would certainly be simpler to keep the naming consistent. Also, because of how older Cobalts stick around, that change would likely be semi-permanent.

---

Yeah let's discuss in the doc—my initial impressions after speaking to Matt and Vartan were that it would be a little annoying for them and they'd probably prefer the h5vcc prefix sticking around, but having a more official discussion out in the open will be good.